static_src/test/global_setup.js

```diff
@@ -0,0 +1,242 @@
+
+import TestUtils from 'react/lib/ReactTestUtils';
+
+Function.prototype.bind = Function.prototype.bind || function (thisp) {
+  var fn = this;
+  return function () {
+    return fn.apply(thisp, arguments);
+  };
+};
+
+var messageFactories = {
+  spy: function(txt) {
+    return function(pass, spy) {
+      return messageUtils.expectedSpy(pass, spy, txt) + '.';
+    };
+  },
+  spyWithCallCount: function(txt) {
+    return function(pass, spy, otherArgs) {
+      return messageUtils.expectedSpy(pass, spy, txt) + '. ' +
+        messageUtils.callCount(spy) + '.';
+    };
+  },
+  spyWithOtherArgs: function(txt) {
+    return function(pass, spy, otherArgs) {
+      return messageUtils.expectedSpy(pass, spy, txt) + ' ' +
+        messageUtils.otherArgs(otherArgs);
+    };
+  }
+};
+
+var messageUtils = {
+  expectedSpy: function(pass, spy, txt) {
+    var not = (pass ? 'not ' : '');
+    var printf = spy.printf || sinon.spy.printf;
+    return printf.call(spy, 'Expected spy "%n" %1%2', not, txt);
+  },
+  callCount: function(spy) {
+    var printf = spy.printf || sinon.spy.printf;
+    return printf.call(spy, '"%n" was called %c');
+  },
+  otherArgs: function(otherArgs) {
+    if (!otherArgs || !otherArgs.length) {
+      return '';
+    } else if (otherArgs.length > 1) {
+      return jasmine.pp(otherArgs);
+    } else {
+      return jasmine.pp(otherArgs[0]);
+    }
+  }
+};
+
+var sinonMatchers = [
```

Comment:
why do we have to translate between sinon and jasmine matchers? why not only have one?